core/src/main/java/org/mule/runtime/core/internal/execution/ModuleFlowProcessingPhase.java

```diff
@@ -105,6 +105,7 @@ public ModuleFlowProcessingPhase(PolicyManager policyManager) {
   @Override
   public void initialise() throws InitialisationException {
     final ErrorTypeRepository errorTypeRepository = muleContext.getErrorTypeRepository();
+    componentLocator = muleContext.getConfigurationComponentLocator();
```

Comment:
Have you looked for all other uses of `getConfigurationComponentLocator` to ensure they are not done as part of message processing?